pkg/splunk/enterprise/searchheadcluster.go

```diff
@@ -667,3 +673,64 @@ func getSearchHeadClusterList(ctx context.Context, c splcommon.ControllerClient,
 
 	return objectList, nil
 }
+
+// isSearchHeadReadyForUpgrade checks if SearchHeadCluster can be upgraded if a version upgrade is in-progress
+// No-operation otherwise; returns bool, err accordingly
+func isSearchHeadReadyForUpgrade(ctx context.Context, c splcommon.ControllerClient, cr *enterpriseApi.SearchHeadCluster) (bool, error) {
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("isSearchHeadReadyForUpgrade").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+	eventPublisher, _ := newK8EventPublisher(c, cr)
+
+	// check if a MonitoringConsole is attached to the instance
+	monitoringConsoleRef := cr.Spec.MonitoringConsoleRef
+	if monitoringConsoleRef.Name == "" {
+		return true, nil
+	}
+
+	namespacedName := types.NamespacedName{
+		Namespace: cr.GetNamespace(),
+		Name:      GetSplunkStatefulsetName(SplunkSearchHead, cr.GetName()),
+	}
+
+	// check if the stateful set is created at this instance
+	statefulSet := &appsv1.StatefulSet{}
+	err := c.Get(ctx, namespacedName, statefulSet)
+	if err != nil && k8serrors.IsNotFound(err) {
```

Comment:
if we just want to return with any err then we do not have to check for IsNotFound

Reply:
Should we instead return a false here if an error occurs and it is not the notFound error?